app/src/test/contrib/stores/ContribAuthStoreTests.ts

```diff
@@ -55,6 +55,19 @@ class ContribAuthStoreTests extends AuthStoreBaseTests<ContribAuthState, Contrib
             expect(contribAuthStore.getState().modellingGroups).to.eql([ group ]);
         });
 
+        it("if the user is a member of a non-existent group, this is ignored", () => {
+            const realGroup = mockModellingGroup({ id: "test.group" });
+
+            alt.bootstrap(JSON.stringify({
+                ContribAuthStore: {
+                    modellingGroupIds: [ realGroup.id, "fake.group" ]
+                }
+            }));
+
+            modellingGroupActions.updateGroups([ realGroup ]);
```

Comment:
I'm not sure I understand this - why is `updateGroups` called with only the real group?

Reply:
So there are two sources of information for the auth store:

1. The auth token, which contains (amongst other things) group *membership" - just the IDs of the groups
1. The `/modelling-groups/` endpoint, which returns full info (i.e. id, name, description) of all groups in the database.

The contrib store combines these two to get an array that contains full info on just the groups the user is a member of.

So here, the `updateGroups` action is mimicking what happens when we fetch all modelling groups from the API. Because the fake group only exists in a role mapping, but not actually in the database, this group would not be present in this list returned from the API.

Reply:
Cool, I suspected something along these lines but struggled to follow the chain of events in the code. But this makes sense.

Reply:
@littledice Sorry it's not clearer in the code!